elastic/kibana
# [Alerting] Restricting alerting health API to users with access to rules

Addresses part of https://github.com/elastic/kibana/issues/102100

## Summary

Added a check to the alerting framework health API to ensure that users have access to at least one rule type before providing health information. Takes advantage of the existing `list_rule_types` API which returns all rule types that a user has access to.

## To Verify
* Create a role that does not have access to any rule type. Create a user for that role
* Verify that when accessing `/api/alerting/_health` as user, user receives `403 Forbidden` response
* Update role to allow access to one or more rule types
* Verify that user with updated role can successfully access `/api/alerting/_health`

### Checklist
